src/atomate2/common/jobs/electrode.py

```diff
@@ -0,0 +1,237 @@
+"""Jobs for electrode analysis."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING, Callable, NamedTuple
+
+from jobflow import Flow, Maker, Response, job
+from pymatgen.analysis.defects.generators import ChargeInterstitialGenerator
+
+if TYPE_CHECKING:
+    from pymatgen.alchemy import ElementLike
+    from pymatgen.analysis.structure_matcher import StructureMatcher
+    from pymatgen.core import Structure
+    from pymatgen.entries.computed_entries import ComputedEntry
+    from pymatgen.io.vasp.outputs import VolumetricData
+
+
+logger = logging.getLogger(__name__)
+
+__author__ = "Jimmy Shen"
+__email__ = "jmmshn@gmail.com"
+
+
+class RelaxJobSummary(NamedTuple):
+    """A summary of a relaxation job."""
+
+    structure: Structure
+    entry: ComputedEntry
+    dir_name: str
+    uuid: str
+
+
+@job
+def get_stable_inserted_structure(
```

Comment:
Does this need to be a job? Could all the contents of this function be put into `ElectrodeInsertionMaker.make`?  Since everything inside this is a job anyway I think that would work.

Reply:
Since the job calls itself recursively I thought it would have to be wrapped in a job.
And I think it has to do this recursively since it doesn't know:
- How many insertions are needed
- How many insertions are topotactic

Side note. I know if `replace` or `addition` is the right thing to use here.  They both seem OK but I don't know if there is some guide to follow.

Reply:
Makes sense, I missed the recursion. `replace` is the right way to go here as that way you can use the outputs of the workflow in another job (see my other comment).